modules/crm_core_match/src/Plugin/crm_core_match/field/Name.php

```diff
@@ -2,41 +2,110 @@
 
 /**
  * @file
- * Implementation of FieldHandlerInterface for name fields.
+ * Contains \Drupal\crm_core_match\Plugin\crm_core_match\field\Name.
  */
 
 namespace Drupal\crm_core_match\Plugin\crm_core_match\field;
+use Drupal\crm_core_contact\ContactInterface;
+use Drupal\field\FieldConfigInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Class for evaluating name fields.
+ *
+ * @CrmCoreMatchFieldHandler (
+ *   id = "name"
+ * )
  */
-class NameMatchField extends FieldHandlerBase {
+class Name extends FieldHandlerBase {
 
   /**
-   * This function is going to add a number of fields based on what the name field is configured to display.
+   * A query factory.
    *
-   * The name field uses text and select fields to set values we will need to pass information into other field handlers
-   * to get the right records to pass back.
-   *
-   * @see DefaultMatchingEngineFieldType::fieldRender()
+   * @var \Drupal\Core\Entity\Query\QueryFactory
+   */
+  protected $queryFactory;
```

Comment:
The base class wrongly assumes that the field matchers need a single query only. Instead it should deal with a queryFactory. Don't break the contracts when subclassing! :-)